### PR TITLE
Updates for dev tools and refactoring

### PR DESCRIPTION
**Pull request by amaslenn**
## Summary
1. Updated `ruff`, `pyright`, `pytest` and `vulture`.
2. Added `pytest-deadfixtures` to avoid keeping unused code in tests.
3. Removed duplicating tests.
4. Fixed warning string for missing strategy.
5. Make sure that tests have no unused code using `coverage` tool.
6. Speed up `test_acceptance.py` by mocking `sleep`.

## Test Plan
CI.

## Additional Notes
—

### CODE REVIEW

File: tests/test_acceptance.py [hunk 1]
```diff
@@ -84,93 +85,6 @@ def test_slurm(tmp_path: Path, scenario: Dict):
...
-
-    result = installer.uninstall([test_template_failure])
-
-    assert not result.success
-    assert result.message == "Some test templates failed to uninstall."
-
-    # Check if the template is still installed
-    assert installer.is_installed([test_template_failure])
-
-
```
Comment (TaekyungHeo) [inline tests/test_acceptance.py:173]: Could you please let me know why we are deleting these tests?

Comment (amaslenn) [reply tests/test_acceptance.py:173]: These tests are for some reason duplicated in `tests/test_base_installer.py`

**Review by TaekyungHeo — APPROVED** [review]

**Comment by TaekyungHeo** [comment]
@amaslenn , should we update CloudAIX settings accordingly for consistency?

**Comment by amaslenn** [comment]
> should we update CloudAIX settings accordingly for consistency?

Yes, you are right, I'll do that.

**Pull request merged**